File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/GestureSettings.java
package io.capawesome.capacitorjs.plugins.maplibre.classes;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;

/**
 * The gestures the user can perform on a map. A gesture that is not provided is left unchanged.
 */
public class GestureSettings {

    @Nullable
    private final Boolean pan;

    @Nullable
    private final Boolean rotate;

    @Nullable
    private final Boolean tilt;

    @Nullable
    private final Boolean zoom;

    public GestureSettings(@NonNull JSObject object) {
        this.pan = object.getBoolean("pan", null);
        this.rotate = object.getBoolean("rotate", null);
        this.tilt = object.getBoolean("tilt", null);
        this.zoom = object.getBoolean("zoom", null);
    }

    @Nullable
    public Boolean getPan() {
        return pan;
    }

    @Nullable
    public Boolean getRotate() {
        return rotate;
    }

    @Nullable
    public Boolean getTilt() {
        return tilt;
    }

    @Nullable
    public Boolean getZoom() {
        return zoom;
    }
}

File: packages/maplibre/ios/Plugin/Classes/GestureSettings.swift
import Capacitor
import Foundation

public class GestureSettings {
    let pan: Bool?
    let rotate: Bool?
    let tilt: Bool?
    let zoom: Bool?

    init(pan: Bool?, rotate: Bool?, tilt: Bool?, zoom: Bool?) {
        self.pan = pan
        self.rotate = rotate
        self.tilt = tilt
        self.zoom = zoom
    }

    static func fromJSObject(_ object: JSObject?) -> GestureSettings {
        return GestureSettings(
            pan: object?["pan"] as? Bool,
            rotate: object?["rotate"] as? Bool,
            tilt: object?["tilt"] as? Bool,
            zoom: object?["zoom"] as? Bool
        )
    }
}
